File: src/Homework1/Homework1/Calculator.cs
using System;
using System.Diagnostics.Contracts;

namespace Homework1
{
    public static class Calculator
    {
        public static int Calculate(string operation, int val1, int val2)
        {
            switch (operation)
            {
                case "+":
                    return val1 + val2;
                case "-":
                    return val1 - val2;
                case "*":
                    return val1 * val2;
                case "/":
                    if (val2 == 0)
                    {
                        Console.WriteLine("Error: Val 2 was 0");
                        return Int32.MaxValue;
                        //throw new DivideByZeroException("Val 2 was 0");
                    }

                    return val1 / val2;
                default:
                    return 0;
            }
        }
    }
}

File: src/HW6/Calculator.fs
module HW6.Calculator

open HW6.Expression

let Calculate (val1: decimal) operation (val2: decimal) =
        match operation with
        | "sum" -> Ok (val1 + val2)
        | "dif" -> Ok (val1 - val2)
        | "mult" -> Ok (val1 * val2)
        | "div" ->
        if val2 = 0m then
           Error "Divide by zero"
        else
           Ok (val1 / val2)
        | _ -> Error "Undefined operation"
        
    
let CalculateExpression (expression: Expression) =
    Calculate expression.V1 expression.Operation expression.V2
